### PR TITLE
feat: handle delegation subgraphs and hide subgraph API key in env

### PR DESCRIPTION
**Pull request by ChaituVR**
Primary goal is to hide Subgraph API key that is public here https://github.com/snapshot-labs/snapshot.js/blob/master/src/delegationSubgraphs.json#L2

### Summary
- Move delegation subgraphs from https://github.com/snapshot-labs/snapshot.js/blob/master/src/delegationSubgraphs.json
- Hide API key in env

### How to test
- Add `SUBGRAPH_KEY=0f15b42bdeff7a063a4e1757d7e2f99e` to .env
- If we send a request to a new URL format, for example `'http://localhost:3000/delegation/1` or `'http://localhost:3000/delegation/10` should work,  try the below curl:
```bash
curl --location 'http://localhost:3000/delegation/1' \
--header 'accept: application/json' \
--header 'Content-Type: application/json' \
--data '{"query":"query { delegations (where: {delegate: \"0x24f15402c6bb870554489b2fd2049a85d75b982f\"}, first: 1000) { delegator space } }","variables":{}}'
```
- Everything should work the same as before, for example `http://localhost:3000/api.thegraph.com/subgraphs/name/snapshot-labs/snapshot-goerli` , try below curl
```bash
curl --location 'http://localhost:3000/api.thegraph.com/subgraphs/name/snapshot-labs/snapshot-optimism' \
--header 'accept: application/json' \
--header 'Content-Type: application/json' \
--data '{"query":"query { delegations (where: {delegate: \"0x24f15402c6bb870554489b2fd2049a85d75b982f\"}, first: 1000) { delegator space } }","variables":{}}'
```

### To do:
- [x] Add `SUBGRAPH_KEY` to DO

### CODE REVIEW

**Comment by wa0x6e** [comment]
Both test `curl` commands are giving me:

```
{"errors":[{"message":"No query provided"}]}%
```

**Comment by ChaituVR** [comment]
> Both test `curl` commands are giving me:
> 
> ```
> {"errors":[{"message":"No query provided"}]}%
> ```

Ahh missing a header 😄 

try these:
```bash
curl --location 'http://localhost:3000/delegation/1' \
--header 'accept: application/json' \
--header 'Content-Type: application/json' \
--data '{"query":"query { delegations (where: {delegate: \"0x24f15402c6bb870554489b2fd2049a85d75b982f\"}, first: 1000) { delegator space } }","variables":{}}'
```

```bash
curl --location 'http://localhost:3000/api.thegraph.com/subgraphs/name/snapshot-labs/snapshot-optimism' \
--header 'accept: application/json' \
--header 'Content-Type: application/json' \
--data '{"query":"query { delegations (where: {delegate: \"0x24f15402c6bb870554489b2fd2049a85d75b982f\"}, first: 1000) { delegator space } }","variables":{}}'
```

File: src/api.ts [hunk 1]
```diff
@@ -25,8 +25,7 @@ router.post('/*', async (req, res) => {
   if (!url) return subgraphError(res, 'No subgraph URL provided', 400);
   if (!query) return subgraphError(res, 'No query provided', 400);
 
-  url = url.startsWith('http') ? url : `https://${url}`;
-
+  url = buildURL(url);
```
Comment (wa0x6e) [inline src/api.ts:28]:
```suggestion
  try {
    url = buildURL(url);
  } catch (error: any) {
    return subgraphError(res, error.message, 400);
  }
```

File: src/utils.ts [hunk 2]
```diff
@@ -40,3 +41,12 @@ const httpsAgent = new https.Agent({ keepAlive: true });
 const fetchWithKeepAlive = (uri: any, options: any = {}) => {
   return fetch(uri, { agent: httpsAgent, ...options });
 };
+
+export function buildURL(url: string): string {
+  if (url.startsWith('http')) return url;
+  if (url.startsWith('delegation')) {
+    const network = url.split('/')[1];
+    if (network && delegationSubgraphs[network]) return delegationSubgraphs[network];
```
Comment (wa0x6e) [inline src/utils.ts:49]:
```suggestion
    if (!delegationSubgraphs[network]) throw new Error('Invalid delegation network');

    return delegationSubgraphs[network];
```

**Review by wa0x6e — APPROVED** [review]
tAck, left some suggestion to return an explicit error, when the delegation network is not supported

File: src/api.ts (hunk 1 above)
Comment (ChaituVR) [reply src/api.ts:28]: But i think we still need to support other URLs than delegation URLs. maybe we get more subgraphs in the future?

If we want to remove support for that, we have to do it after merging this PR and changing it here https://github.com/snapshot-labs/snapshot.js/blob/master/src/delegationSubgraphs.json else we end up with failing requests

**Pull request merged**